internal/btf/btf.go

```diff
@@ -507,13 +568,19 @@ func (s *Spec) FindType(name string, typ interface{}) error {
 		return fmt.Errorf("%T does not satisfy Type interface", typ)
 	}
 
+	types, err := s.AnyTypesByName(name)
+	if err != nil {
+		return err
+	}
+
 	var candidate Type
-	for _, typ := range s.namedTypes[essentialName(name)] {
+	for _, typ := range types {
 		if reflect.TypeOf(typ) != wanted {
 			continue
 		}
 
-		// Match against the full name, not just the essential one.
+		// Match against the full name, not just the essential one
+		// in case the type being looked up is a struct flavor.
 		if typ.TypeName() != name {
```

Comment:
Doesn't this mean that `TypeByName("foo", ...)` wont return `foo___flavour`?

Reply:
True, but if I understand it correctly, this has always been the case. What should be the behaviour of `TypeByName("foo", ...)` if e.g. `foo`, `foo___v46` and `foo___v51` are present?